taskweaver/utils/embedding.py

```diff
@@ -0,0 +1,86 @@
+from injector import inject
+from openai import OpenAI
+from openai.lib.azure import AzureOpenAI
+
+from taskweaver.config.module_config import ModuleConfig
+from taskweaver.llm import LLMModuleConfig
+
+
+class EmbeddingModuleConfig(ModuleConfig):
+    def _configure(self) -> None:
+        self._set_name("embedding_model")
+        self.embedding_model_type = self._get_enum(
+            "embedding_model_type",
+            ["sentence_transformer", "openai"],
+            "sentence_transformer",
+        )
+        self.embedding_model_candidates = {
+            "sentence_transformer": [
+                "all-mpnet-base-v2",
+                "multi-qa-mpnet-base-dot-v1",
+                "all-distilroberta-v1",
+                "all-MiniLM-L12-v2",
+                "multi-qa-MiniLM-L6-cos-v1",
+            ],
+            "openai": [
+                "text-embedding-ada-002",
+            ],
+        }
+        self.embedding_model = self._get_enum(
+            "embedding_model",
+            self.embedding_model_candidates[self.embedding_model_type],
+            self.embedding_model_candidates[self.embedding_model_type][0],
+        )
+
+
+class EmbeddingGenerator:
+    @inject
+    def __init__(self, config: EmbeddingModuleConfig, llm_config: LLMModuleConfig):
+        self.config = config
+        self.llm_config = llm_config
+
+        if self.config.embedding_model_type == "sentence_transformer":
+            try:
+                from sentence_transformers import SentenceTransformer
+
+                self.embedding_model = SentenceTransformer(self.config.embedding_model)
+            except Exception:
+                raise Exception(
+                    "Package sentence_transformers is required for using embedding. "
+                    "Please install it using pip install sentence_transformers",
+                )
+        elif self.config.embedding_model_type == "openai":
```

Comment:
Seems the initialization of this part is rather overlapped with the LLM module, not sure whether we can somehow have these two part merged into one place? For example, the AzureAD is not supported in this part.